socorro/processor/mozilla_transform_rules.py

```diff
@@ -391,89 +391,23 @@ def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
         return True
 
 
-def setup_product_id_map(config, local_config, args_unused):
-    database_connection = local_config.database_class(local_config)
-    transaction = local_config.transaction_executor_class(
-        local_config,
-        database_connection
-    )
-    sql = (
-        "SELECT product_name, productid, rewrite FROM "
-        "product_productid_map WHERE rewrite IS TRUE"
-    )
-    product_mappings = transaction(
-        execute_query_fetchall,
-        sql
-    )
-    product_id_map = {}
-    for product_name, productid, rewrite in product_mappings:
-        product_id_map[productid] = {
-            'product_name': product_name,
-            'rewrite': rewrite
-        }
-    return product_id_map
-
-
 class ProductRewrite(Rule):
-    required_config = Namespace()
-    required_config.add_option(
-        'database_class',
-        doc="the class of the database",
-        default='socorro.external.postgresql.connection_context.'
-                'ConnectionContext',
-        from_string_converter=str_to_python_object,
-        reference_value_from='resource.postgresql',
-    )
-    required_config.add_option(
-        'transaction_executor_class',
-        default="socorro.database.transaction_executor."
-                "TransactionExecutorWithInfiniteBackoff",
-        doc='a class that will manage transactions',
-        from_string_converter=str_to_python_object,
-        reference_value_from='resource.postgresql',
-    )
-
-    required_config.add_aggregation(
-        'product_id_map',
-        setup_product_id_map
-    )
-
     def __init__(self, config):
         super(ProductRewrite, self).__init__(config)
-        self.product_id_map = setup_product_id_map(
-            config,
-            config,
-            None
-        )
+        self.product_map = {
```

Comment:
should this be a:
* `CONSTANT`
* hoist it out to a class variable like `required_confid`

Reply:
I was thinking about that. I want to think about this some more and see if there are other tables in similar situations.

Reply:
I would make it a "constant" in this module for now. That makes it easier to re-use elsewhere.

I can't think of any good reasons to make it a configuration option at this time. If something ever popped up, we can change it then.

Reply:
I have another comment elsewhere in this PR, but to make it easier, I'm going back on my "make it a module-level constant" and changing my mind to "make it a class attribute since that's what we do with other rules".